### ISSUE TITLE
Use the internal Tumblr SVC API to access hidden or explicit blogs.

### ISSUE THREAD

**Issue by MrEldritch** [bbolli/tumblr-utils#161]
After some research and experimentation, I've managed to find an alternative API that can be used to access hidden or explicit Tumblrs and posts that require a login to view. I've written a proof-of-concept for how to access the API and retrieve a JSON of very similar format to what the regular API uses [here](https://gist.github.com/MrEldritch/a32467514a9052fad9b20784ae060800).

The resulting JSON is almost, but not quite, identical to the usual format; the primary differences are simply that some fields are given different names, and that the "blog" field that gives general information about the Tumblr is given as part of each "post" rather than as a separate top-level field. (Also, it doesn't include the total number of posts for that blog). There are a few other differences, but for the most part all the information is still there and in the same place; so while a separate parser would need to be written, it would not be _very_ separate. (It's also quite straightforward to detect whether a tumblr is hidden or not, and therefore which parser you'd need to use - you just try to access it at the normal blogname.tumblr.com URL, and see if it redirects you to a URL containing "login_required" or "dashboard/blog/".)

I've been working on trying to fully integrate this functionality myself, but it's been slow going as I'm still working out how your own script works exactly. (It doesn't help that I'm primarily a Python 3 user, myself); I could use some help.

**Comment by MrEldritch** [comment]
As an example, [here's](https://gist.github.com/MrEldritch/8e4f464a7503e07ab11352f13e52a64d) what the JSON looks like when you request the first 50 posts using the SVC API via the proof-of-concept method I included, and [here's](https://gist.github.com/MrEldritch/b304878dcaf9fcce080dc2f4e296e18e) what you get from the ordinary Tumblr v2 API you're already using.

I have not yet worked out a good solution for archiving *liked* posts from hidden or explicit tumblrs.

**Comment by MrEldritch** [comment]
Actually, nevermind, archiving liked posts from hidden or explicit tumblrs is piss-easy; you can just use the ordinary v2 API url.

**Comment by johanneszab** [comment]
You could at least have said that you just re-wrote [code from TumblThree](https://github.com/johanneszab/TumblThree/blob/08059233c050300c2d7472ce4df0fe649e86a82b/src/TumblThree/TumblThree.Applications/Services/LoginService.cs#L98) into python instead of claiming you managed to find an alternative to the api. You even have the same ["user name" suggestions](https://gist.github.com/MrEldritch/a32467514a9052fad9b20784ae060800#file-tumblr_login_proofofconcept-py-L68) returned from the Tumblr SVC, which I highly doubt will ever be suggested a second time, next to the same parameter set, method names, redirection detection and user agent and all the other suggestions here.

No hard feelings, it's all open source, and I'd like to see to spread, but I actually sat there a few days figuring out what request to send and how to answer them.

**Comment by MrEldritch** [comment]
@johanneszab You're right on that count. To be clear, this was just a follow-on from [some discussion on another issue](https://github.com/bbolli/tumblr-utils/issues/128#issuecomment-444636483) where I'd been explicitly stating that I was trying to reverse-engineer how TumblThree did it; I simply felt I'd put together enough progress that I wanted to make a new issue to boost it in visibility. 

I will say that this also took _me_ several days, because I had literally never read a line of C# _or_ worked with Python 2 _or_ done anything involving HTTP requests before, so trying to figure out how TumblThree and tumblr-utils actually worked well enough to figure out *what* the relevant bits were and how to translate them ended up filling up quite a few sleepless nights. 

...although if I'd actually spent those nights sleeping, it probably would have occurred to me that I could have just _asked you how it worked,_ which I have only just now realized.

**Comment by MrEldritch** [comment]
@johanneszab ...also, that proof-of-concept code originally had some comments literally reading `#This bit is pure voodoo to me; I just copied these requests from TumblThree's code. 
#You need to make these requests to pick up the necessary cookies; I don't know why it works, but it does.`

I thought those were still in there, but it looks like I accidentally deleted them while I was cleaning up and re-organizing that code into its own procedure.
